vllm/entrypoints/openai/serving_chat.py

```diff
@@ -668,6 +669,10 @@ async def chat_completion_full_generator(
                             arguments=output.text))
                     ])
 
+                if isinstance(tokenizer, MistralTokenizer):
+                    for tool_call in message.tool_calls:
+                        tool_call.id = generate_valid_mistral_tool_id()
```

Comment:
I just noticed that there is a `MistralToolCall` class that overrides the id generation. It would be a bit cleaner to just use that class, eg. could make this change above:
```
                tool_call_class = MistralToolCall if isinstance(tokenizer, MistralTokenizer) else ToolCall
                message = ChatMessage(
                    role=role,
                    content="",
                    tool_calls=[
                        tool_call_class(function=FunctionCall(
                            name=request.tool_choice.function.name,
                            arguments=output.text))
                    ])
```